pkg/cmd/start/main.go

```diff
@@ -182,11 +182,14 @@ func start(cmd *cobra.Command, args []string) {
 	}
 
 	// Create Service object to expose the metrics port.
-	var operatorService *corev1.Service
-	if operatorService, err = metrics.ExposeMetricsPort(ctx, viper.GetInt32("metrics-port")); err != nil {
-		log.Fatal(err)
-	} else if err = setOwnerReference(operatorService); err != nil {
+	operatorService, err := metrics.ExposeMetricsPort(ctx, viper.GetInt32("metrics-port"))
+	if err != nil {
 		log.Fatal(err)
+	} else if operatorService != nil {
```

Comment:
This check was missing there. The `ExposeMetricsPort` returns `nil, nil` if the code does not run inside the k8s cluster.